src/back/spv/block.rs

```diff
@@ -888,6 +888,127 @@ impl<'w> BlockContext<'w> {
+                                    vector_size: Some(size),
+                                    kind: crate::ScalarKind::Uint,
+                                    width,
+                                    pointer_space: None,
+                                }));
+
+                                self.temp_list.clear();
+                                self.temp_list.resize(
+                                    size as _,
+                                    self.writer.get_constant_scalar(uint, width),
+                                );
+
+                                let id = self.gen_id();
+                                block.body.push(Instruction::constant_composite(
+                                    ty,
+                                    id,
+                                    &self.temp_list,
+                                ));
+
+                                (ty, id)
+                            }
+                            crate::TypeInner::Scalar { width, .. } => (
+                                self.get_type_id(LookupType::Local(LocalType::Value {
+                                    vector_size: None,
+                                    kind: crate::ScalarKind::Uint,
+                                    width,
+                                    pointer_space: None,
+                                })),
+                                self.writer.get_constant_scalar(uint, width),
+                            ),
+                            _ => unreachable!(),
+                        };
+
+                        let mut arg_id = arg0_id;
+                        if let Some(crate::ScalarKind::Uint) = arg_scalar_kind {
+                            arg_id = self.gen_id();
+                            block.body.push(Instruction::unary(
+                                spirv::Op::Bitcast,
+                                int_type_id,
+                                arg_id,
+                                arg0_id,
+                            ));
+                        }
+
+                        block.body.push(Instruction::ext_inst(
+                            self.writer.gl450_ext_inst_id,
+                            spirv::GLOp::FindILsb,
+                            int_type_id,
+                            id,
+                            &[arg_id],
+                        ));
+
+                        let cast_id = self.gen_id();
+                        block.body.push(Instruction::unary(
+                            spirv::Op::Bitcast,
+                            uint_type_id,
+                            cast_id,
+                            id,
+                        ));
+
+                        let min_id = self.gen_id();
+                        block.body.push(Instruction::ext_inst(
+                            self.writer.gl450_ext_inst_id,
+                            spirv::GLOp::UMin,
+                            uint_type_id,
```

Comment:
I think we can use `result_type_id` here directly, removing the need for the bitcast below.